src/librustdoc/html/render/print_item.rs

```diff
@@ -2267,6 +2271,53 @@ fn document_non_exhaustive<'a>(item: &'a clean::Item) -> impl fmt::Display + 'a
     })
 }
 
+fn document_default_header(
+    cx: &mut Context<'_>,
+    it: &clean::Item,
+    variants: &IndexVec<VariantIdx, clean::Item>,
+) -> String {
+    let tcx = cx.tcx();
+    let cache = cx.cache();
+    let v: Vec<&Impl> = cx.shared.all_impls_for_item(it, it.item_id.expect_def_id());
+
+    let default_fqp = vec![sym::core, kw::Default, sym::Default];
+    let mut label = "".to_string();
+    v.into_iter().for_each(|i| {
+        if let clean::ImplItem(ref impl_) = *i.impl_item.kind && let Some(trait_) = &impl_.trait_ {
+            let did = trait_.def_id();
+
+            let fqp = if let Some(&(ref fqp, _)) = cache.paths.get(&did) {
```

Comment:
I'm not sure if I need this, or line 2292 is enough.
And generally the whole logic to be sure we are implementing core::default::Default.
There might be some helpers there but I couldn't find it.

Reply:
Right, it does look more convoluted than it probably needs to be. I won't suggest any changes for that rn, see https://github.com/rust-lang/rust/pull/115575#discussion_r1369328054